pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -142,6 +148,40 @@ def _math_define_validator(value, values):
     return output
 
 
+class _ChunkResizer:
+    """The only purpose of this class is to resize the chunk size of the instrument adapter.
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+    Note.
+    Only if the new chunk size is bigger than the current chunk size, it is resized. """
+
+    def __init__(self, adapter, chunk_size):
+        """ Just initialize the object attributes.
+        :param: adapter of the instrument. This is usually accessed through the
+        Instrument::adapter attribute.
+        :chunk_size: new chunk size (int).
+        """
+        self.adapter = adapter
+        self.old_chunk_size = None
+        self.new_chunk_size = int(chunk_size)
+
+    def __enter__(self):
+        """ Only resize the chunk size if the adapter support this feature"""
+        if (hasattr(self.adapter, "connection")
+                and self.adapter.connection is not None
+                and hasattr(self.adapter.connection, "chunk_size")):
+            self.old_chunk_size = self.adapter.connection.chunk_size
+            if self.new_chunk_size > self.old_chunk_size:
+                self.adapter.connection.chunk_size = self.new_chunk_size
```

Comment:
If you pull setting the old_chunk_size into the if here, in the exit method you only have to test for ` self.old_chunk_size is not None` -- the other conditions must have been true to even reach the point where old_chunk_size gets set.
Also, you avoid the bug/case where old_chunk_size is set, but then the if here is false, so the setting back is actually unnecessary.

Reply:
Very good point. Implemented in commit d34f0ae535bf2b2aeb76225f7ba91a982e286f88